Combat/Source/Points.cpp

```diff
@@ -0,0 +1,48 @@
+#include "Points.hpp"
+
+Points::Points(const uint maxPoints)
+{
+   _maxPoints = maxPoints;
+   _currentPoints = maxPoints;
+}
+
+void Points::remove(const uint amount)
+{
+   if(amount > _currentPoints)
+   {
+      _currentPoints = 0;
+   }
+
+   _currentPoints -= amount;
+}
+
+void Points::add(const uint amount)
+{
+   if(_currentPoints + amount > _maxPoints)
+   {
+      _currentPoints += amount;
+   }
+
+   _currentPoints += amount;
+}
```

Comment:
Według mnie to nie zadziała, bo czemu mimo przekroczenia 'bariery' _maxPoints chcesz dodać amount do _currentPoints? Moim zdaniem po gdy przy dodawaniu przekraczamy bariere, to _currentPoints = _maxPoints